src/Query/QueryBuilder.php

```diff
@@ -449,7 +450,7 @@ public function getParameterTypes(): array
      *
      * @param int|string $key The key of the bound parameter type
      */
-    public function getParameterType(int|string $key): string|ParameterType|Type|ArrayParameterType
+    public function getParameterType(int|string $key): string|ParameterType|Type|ArrayParameterType|ArrayType
```

Comment:
BC break?

Reply:
My idea was that `ArrayType` is a `Type`.

Reply:
What would it do to convert a PHP value (an array, probably?) to a database value?

Reply:
It will delegate the conversion of each element of the array to its element type. We can start simple and accept only `Type` in the constructor to prove the idea:
```php
public function __construct(private readonly Type $elementType)
```

Reply:
When would it be necessary to use the `ArrayType` in this way?

Reply:
When building the query [here](https://github.com/doctrine/orm/blob/a67f677747178ebb4d878c2626daa12e10f18fd8/src/Persisters/Entity/BasicEntityPersister.php#L924). If this code builds the parameter types as `[new ArrayType('rot13'), 'rot13']`, the DBAL should be able to apply `rot13` to the array elements similar to how it does with a single value.